deepfence_worker/cronjobs/notification.go

```diff
@@ -307,22 +307,97 @@ func processIntegration[T any](ctx context.Context, task *asynq.Task, integratio
 		FieldsValues: map[string][]interface{}{"status": {utils.ScanStatusSuccess}},
 	}
 
+	scansList := []model.ScanInfo{}
+	scanType, ok := utils.DetectedNodeScanType[integrationRow.Resource]
+	if !ok {
+		log.Error().Msgf("Unsupported scan type in integration: %v", integrationRow)
+		return errors.New("Unsupported scan type in integration")
+	}
+
+	scanFilter := reporters_search.SearchFilter{
+		Filters: filters.FieldsFilters,
+	}
+
+	filterMap := make(map[string][]interface{})
+
+	containerFilterMap := make(map[string][]interface{})
+	containerNames := make([]interface{}, len(filters.ContainerNames))
+	for i, v := range filters.ContainerNames {
+		containerNames[i] = v
+	}
+
+	var nodeIDs, nodeTypes []interface{}
+	nodeTypeSet := make(map[string]bool)
+	for _, v := range filters.NodeIds {
+		if v.NodeType == "container" {
+			continue
+		}
+		nodeIDs = append(nodeIDs, v.NodeID)
+		if _, ok := nodeTypeSet[v.NodeType]; !ok {
```

Comment:
Can we encapsulate the whole inside a nice `integrationFilters2SearchScanReqs` function?